workspaces/rbac/plugins/rbac/src/components/CreateRole/MemberSelectionDialog.tsx

```diff
@@ -0,0 +1,305 @@
+/*
+ * Copyright 2025 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { useState, useEffect, useMemo } from 'react';
+import {
+  GroupEntity,
+  stringifyEntityRef,
+  UserEntity,
+} from '@backstage/catalog-model';
+import { Table, TableColumn, Progress } from '@backstage/core-components';
+import {
+  Button,
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogTitle,
+  Tabs,
+  Tab,
+  Box,
+  makeStyles,
+  createStyles,
+  Theme,
+  IconButton,
+} from '@material-ui/core';
+import CloseIcon from '@mui/icons-material/Close';
+import { TranslationFunction } from '@backstage/core-plugin-api/alpha';
+import { rbacTranslationRef } from '../../translations';
+import { SelectedMember } from './types';
+import { MemberEntity } from '../../types';
+import {
+  getChildGroupsCount,
+  getMembersCount,
+  getParentGroupsCount,
+} from '../../utils/create-role-utils';
+
+const useStyles = makeStyles((theme: Theme) =>
+  createStyles({
+    dialogCloseButton: {
+      position: 'absolute',
+      right: theme.spacing(1),
+      top: theme.spacing(1),
+      color: theme.palette.grey[500],
+    },
+  }),
+);
+
+const getDescription = (
+  member: MemberEntity,
+  t: TranslationFunction<typeof rbacTranslationRef.T>,
+) => {
+  const memberCount = getMembersCount(member);
+  const parentCount = getParentGroupsCount(member);
+  const childCount = getChildGroupsCount(member);
+
+  return member.kind === 'Group'
+    ? [
+        memberCount > 0
+          ? t('common.membersCount' as any, { count: memberCount })
+          : '',
+        parentCount > 0
+          ? t('common.parentGroupCount' as any, { count: parentCount })
+          : '',
+        childCount > 0
+          ? t('common.childGroupsCount' as any, { count: childCount })
+          : '',
+      ]
+        .filter(Boolean)
+        .join(', ')
+    : undefined;
+};
+
+const entityToSelectedMember = (
+  entity: MemberEntity,
+  t: TranslationFunction<typeof rbacTranslationRef.T>,
+): SelectedMember => {
+  const ref = stringifyEntityRef(entity);
+  return {
+    id: entity.metadata.etag ?? ref,
+    label: entity.spec?.profile?.displayName ?? entity.metadata.name,
```

Comment:
Could we use the [Presentation API ](https://github.com/backstage/backstage/pull/17474)to make this better? If not the `metadata.title` would be worth try before `metadata.name`